kolibri/core/content/generate_fixtures.py

```diff
@@ -0,0 +1,299 @@
+from le_utils.constants import content_kinds
+from le_utils.constants import format_presets
+
+content_kinds_choices = content_kinds.choices
+format_presets_choices = format_presets.choices
+
+# names of all apps in interest, why?
+apps_names = ["content","lessons","exams", "logger"]
+# because later i want to make this script more flexible to be able to generate fixtures for any app
+# I can imagine for example that the app name is provided as an argument when we run this script from the terminal 
+# note : apps_names is not used yet
+
+current_app_name = KolibriContentConfig.label
+
+
+# get current app models names (content app in this case) that will be dumped
+def GetAppModels(app_name):
+
+   # first way, extracts only the base (main) models
+    models_names = [f"{app_name}.{model.__name__}" for model in list(apps.get_app_config(app_name).get_models())]
+   
+   # second way, extracts all app models even if it doesn't explicilty inherit from models.Model (e.g. in content app : contentnode_has_prerequisite, contentnode_has_prerequisite,etc..)
+    all_models_names = [f"{app_name}.{model}" for model in dict(apps.all_models[app_name])]
+
+   # which one should we use ? (i.e which models will be dumped ? the ones extraced through the first or the second way)
+    
+    return models_names
+
+
+# populating the current app models 
+
+app_models = GetAppModels(current_app_name)
+
+
+
+def generate_random_id():
+    import uuid
+    return uuid.uuid4().hex 
+
+
+
+
+
+content_kind_to_file_preset = {
+    "topic": ["topic_thumbnail"],
+    "video" : ["video_subtitle","video_thumbnail","low_res_video","high_res_video"],
+    "audio": ["audio","audio_thumbnail"],
+    "exercise": ["exercise","exercise_thumbnail","qti"],
+    "document": ["document","document_thumbnail"],
+    "html5": ["html5_zip","html5_thumbnail"],
+    "slideshow":["slideshow_manifest","slideshow_image"],
+    "h5p" : ["h5p","audio_thumbnail"],
+}
+
+
+
+# code used to generate the following file_preset_to_localfile_extension mapper :
+#     mapper = {}       
+#     for form in format_presets:
+#         mapper[form[0]] = LocalFile.objects.filter(id__in=[file.local_file.id for file in File.objects.filter(preset=form[0])])[0].extension
+
+file_preset_to_localfile_extension={
+    "high_res_video":"mp4",
+    "low_res_video":"mp4",
+    "video_thumbnail":"png",
+    "audio_thumbnail":"png",
+    "exercise_thumbnail":"png",
+    "topic_thumbnail":"png",
+    "html5_thumbnail":"png",
+    "document_thumbnail":"jpg",
+    "slideshow_image":"jpg",
+    "html5_zip":"zip",
+    "qti":"zip",
+    "video_subtitle":"vtt",
+    "audio":"mp3",
+    "document":"pdf",
+    "epub":"epub",
+    "exercise":"perseus",
+    "h5p":"h5p",
+    "slideshow_manifest":"json",
+}
+ 
+
+
+# any generated object will be added here to get deleted later after dumping
+generated_objects = set()
+
+
+
+# keeps track of used (previously generated) languages (the ones that will be dumped) 
+Used_Languages = {}
```

Comment:
A general code style comment - in the Kolibri code base we tend to use snake_case for most Python variables and function names, so this would be `used_languages`.

For class names we use PascalCase.

Reply:
done